qemu/tests/qemu_guest_agent.py

```diff
@@ -752,44 +753,140 @@ def get_allocation_bitmap():
-        add_brige_cmd = "ip link add link lo name lo_brige type bridge"
-        session.cmd(add_brige_cmd)
-
-        interfaces_after_add = self.gagent.get_network_interface()
-
-        bridge_list = [_ for _ in interfaces_after_add if _ not in
-                       interfaces_pre_add]
-        if (len(bridge_list) != 1) or \
-           ("lo_brige" != bridge_list[0]["name"]):
-            test.fail("the interface list info after interface was down "
-                      "was not as expected")
+        error_context.context("Check the available interface name %s"
+                              " via qga." % interface_name, logging.info)
+        if self.params.get("os_type") == "linux":
+            interface_name_guest = utils_net.get_linux_ifname(session_serial, mac_addr)
+        else:
+            interface_name_guest = utils_net.get_windows_nic_attribute(session_serial,
+                                                                       "macaddress",
+                                                                       mac_addr,
+                                                                       "netconnectionid")
+        if interface_name != interface_name_guest:
+            test.fail("Get the wrong interface name, value from qga is: %s; "
+                      "the expected is: %s" % (interface_name, interface_name_guest))
+
+        error_context.context("Check ip address via qga.", logging.info)
+        ip_addr_check(session_serial, mac_addr, ret, index_if, interface_name)
+
+        # create a bridge interface for linux guest and check it from guest agent
+        # disable interface for windows guest and  check it from guest agent
+        if params.get("os_type") == "linux":
+            error_context.context("Create a new bridge in guest and check the"
+                                  "result from qga.", logging.info)
+            add_brige_cmd = "ip link add name br0 type bridge"
+            session_serial.cmd(add_brige_cmd)
+            interfaces_after_add = self.gagent.get_network_interface()
+            interface_num = 0
+            for interface in interfaces_after_add:
+                if interface["name"] == "br0":
+                    break
+                interface_num += 1
+            if interface_num == len(interfaces_after_add):
+                test.fail("The new bridge is not checked from guest agent.")
+            error_context.context("Delete the added bridge.", logging.info)
+            del_brige_cmd = "ip link del br0"
+            session_serial.cmd(del_brige_cmd)
+        else:
+            error_context.context("Set down the interface in windows guest.",
+                                  logging.info)
+            utils_net.disable_windows_guest_network(session_serial, interface_name)
+            ret_after_down = self.gagent.get_network_interface()
+            interface_name_down, index_if_down = get_interface(ret_after_down, mac_addr)
```

Comment:
If `index_if_down` is useless, `interface_name_down = get_interface(ret_after_down, mac_addr)[0]`

Reply:
Thanks update